zokrates_core/src/semantics.rs

```diff
@@ -1411,20 +1411,17 @@ impl<'ast> Checker<'ast> {
                     }),
                 }
             }
-            Expression::FieldConstant(n) => {
-                let bits = n.bits();
-                let required_bits = T::get_required_bits();
-                match bits <= required_bits {
-                    true => Ok(FieldElementExpression::Number(T::from(n)).into()),
-                    false => Err(ErrorInner {
-                        pos: Some(pos),
-                        message: format!(
-                            "Field constant not in the representable range ({} bits)",
-                            required_bits
-                        ),
-                    }),
-                }
-            }
+            Expression::FieldConstant(n) => match n <= T::max_value().to_biguint() {
```

Comment:
Actually now that I read this again, wouldn't it make sense here to implement `impl<T: Field> TryFrom<BigUint> for T` and use this here? Only the error would need to be adjusted with `map_err`.

Right now in `zokrates_field` with have `From`, but it should be removed as it can fail, as this case shows

Reply:
actually yes, makes perfect sense